src/main/java/com/minecolonies/apiimp/initializer/InteractionValidatorInitializer.java

```diff
@@ -307,10 +308,7 @@ public static void init()
           citizen -> citizen.getEntity().isPresent() && citizen.getColony().getRaiderManager().isRaided());
 
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(COM_MINECOLONIES_COREMOD_ENTITY_CITIZEN_SLEEPING),
-          citizen -> citizen.getEntity().isPresent() && citizen.getEntity().get().getCitizenSleepHandler().shouldGoSleep()
-                      && !citizen.getColony().getRaiderManager().isRaided()
-                      && !citizen.getCitizenMournHandler().isMourning()
-                      && !citizen.getEntity().get().getCommandSenderWorld().isRaining());
+          citizen -> citizen.getEntity().isPresent() && citizen.getEntity().get().getDesiredActivity() == DesiredActivity.SLEEP);
```

Comment:
This had the raided checks etc on purpose, because raiding, rain etc also results in desired activity sleep